fix(kong-identity): Update client consumer mapping documentation (#4733)
Kong/developer.konghq.com: app/konnect-platform/kong-identity.md

{% endcomment %}

## Kong Consumer Group claim authorization flow

> To apply plugins to OIDC clients, the Gateway must map the incoming token to a **Consumer** entity. This is handled by the [OIDC plugin](https://developer.konghq.com/plugins/openid-connect/) at request time:

Reviewer: ```suggestion
To apply plugins to OIDC clients, the Gateway must map the incoming token to a **Consumer** entity. This is handled by the [OIDC plugin](/plugins/openid-connect/) at request time:
```

* **Map the Identity**: Ensure the Consumer's `username` or `custom_id` matches a unique claim in the OIDC token. The `sub` claim is included in Kong Identity tokens by default.

> * **Link via Plugin**: Configure the following fields in the OIDC plugin:

Reviewer: ```suggestion
1. Configure the following fields in the OIDC plugin to link requests to a Consumer:
```

    * [`config.consumer_claim`](https://developer.konghq.com/plugins/openid-connect/reference/#config-consumer_claim): The token field to identify the client (e.g., `["sub"]` or `["client_id"]`).
    * [`config.consumer_by`](https://developer.konghq.com/plugins/openid-connect/reference/#config-consumer_by): The Consumer field to match against (e.g., `["username"]` or `["custom_id"]`).
* **Apply Scoped Plugins**:
    * **Consumer-scoped**: Apply plugins directly to the individual Consumer entity.

>     * **Consumer Group-scoped**: Assign mapped Consumers to a [Consumer Group](https://docs.konghq.com/gateway/latest/admin-api/consumer-groups/reference/) to inherit shared policies across multiple clients.

Author: I'm not sure... Claude helped make the changes but I thought I provided/checked all the links 🤔 Sorry about that!

{:.info}

> > You do not need to migrate or create client credentials (like API keys) for these Consumers. The OIDC token serves as the authenticated credential for the mapped Consumer.

Reviewer: ```suggestion
> You don't need to migrate or create client credentials (like API keys) for these Consumers. The OIDC token serves as the authenticated credential for the mapped Consumer.
```
